### PR TITLE
Omit aggregation options converting expressions to legacy

### LINKED ISSUE #36120: legacyExpressionForExpressionClause returns more than just the expression
See [Slack discussion](https://metaboat.slack.com/archives/C0645JP1W81/p1701082998021059).

```
Lib.legacyExpressionForExpressionClause(query, stageIndex, aggregationClause)
// returns ["aggregation-options", ["count"], { "name": "oko", "display-name": "oko" }

// but it should return just ["count"]
```

I also had a case like this:
```
Lib.legacyExpressionForExpressionClause(query, stageIndex, clause)
// returned ['count', null]
```

----

How to reproduce:
1. `git checkout master`
2. Create a question with [named custom expression as aggregation](https://github.com/metabase/metabase/assets/6830683/0ad90cef-518f-4289-94c3-921ac58e5eed) and save it.
3. `git checkout 35947-legacy-expression-clause-filters-and-aggregations` (#36031)
4. Open notebook editor for question created in step 2
5. Add a breakpoint in `ExpressionEditorTextfield.tsx`:
```diff
diff --git a/frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx b/frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx
index 5821a97fbe..b2732ea5f7 100644
--- a/frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx
+++ b/frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/ExpressionEditorTextfield.tsx
@@ -88,7 +88,7 @@ function transformPropsToState(
     query,
     stageIndex,
   } = props;
-
+  debugger;
   const expression = clause
     ? Lib.legacyExpressionForExpressionClause(query, stageIndex, clause)[1] // TODO: remove [1], see https://github.com/metabase/metabase/issues/36120
     : undefined;
```
6. Click the aggregation to edit it
7. Inspect result of `metabase_lib__WEBPACK_IMPORTED_MODULE_6__.legacyExpressionForExpressionClause(query, stageIndex, clause)` ([screenshot](https://github.com/metabase/metabase/assets/6830683/58e44d3a-0604-4f34-9234-0468dba747d8))

### PR DESCRIPTION
**Pull request by metamben**
Fixes #36120 and #36168.

### CODE REVIEW

**Review by kamilmielnik — COMMENTED** [review]
The main issue seems to be fixed :+1: 
I added more context about the `["count", null]` problem in [comments for the original issue](https://github.com/metabase/metabase/issues/36120#issuecomment-1829361821).

**Comment by metamben** [comment]
@kamilmielnik this should fix the `["count" null]` issue.

**Comment by kamilmielnik** [comment]
> @kamilmielnik this should fix the `["count" null]` issue. Looking at the description of #36168, I think that should be fixed too.

I confirm the `["count" null]` issue is fixed, and #36168 is fixed as well :+1:

File: frontend/src/metabase-lib/expression.ts [hunk 1]
```diff
@@ -73,7 +74,7 @@ export function expressionClauseForLegacyExpression(
 export function legacyExpressionForExpressionClause(
   query: Query,
   stageIndex: number,
-  expressionClause: ExpressionClause,
+  expressionClause: ExpressionClause | AggregationClause,
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/expression.ts:77]: We need to support `FilterClause` here as well

Comment (metamben) [reply frontend/src/metabase-lib/expression.ts]: Added `FilterClause` and a test.

**Review by snoe — APPROVED** [review]

**Pull request merged**